### PR TITLE
[sdk/javascript]: improve codegen to significantly reduce size of generated models

### PR DESCRIPTION
**Pull request by Jaguar0625**
problem: generated code is very duplicative
solution: add concept of base class to generator to reduce duplication

### CODE REVIEW

File: sdk/javascript/generator/AbstractTypeFormatter.py [hunk 1]
```diff
@@ -16,6 +16,10 @@ class AbstractTypeFormatter(ABC):
 	def typename(self):
 		raise NotImplementedError('need to override method')
 
+	@property
+	def is_abstract(self):
+		return False
```
Comment (Wayonb) [inline sdk/javascript/generator/AbstractTypeFormatter.py:21]: At first, I thought this referred to the formatted class....maybe ``is_abstract_type/class``?

**Review by Wayonb — APPROVED** [review]
Nice use of inheritance 👍🏾

**Comment by Jaguar0625** [comment]
@Wayonb - let me know if you have any additional comments after the rename 🙏

**Review by Wayonb — APPROVED** [review]
👍🏾

**Pull request merged**